src/Chapter4.hs

```diff
@@ -629,7 +664,11 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+-- andM m1 m2 = (&&) <$> m1 <*> m2
```

Comment:
Ok, nice remark, thanks!

But one question (when monad is []):
```haskell
andM [True, False] [False, True]
```
For Applicative approach:
```haskell
[False,True,False,False]
```
For Monad approach (according to our task with lazy AND):
```haskell
[False,True,False]
```

Is it correct?

Reply:
This is correct. Though, list is rarely used as Monad, so such functions are mainly optimized for `Maybe`s.
It is always safer to control the behaviour of lists and know exactly what kind of implementation is used because as you see, the answer is very different in various cases. 🙂